otter/convergence/service.py

```diff
@@ -531,6 +539,20 @@ def converge_one_group(currently_converging, recently_converged,
         # unexpected errors, so convergence will be retried.
         yield err(None, 'converge-non-fatal-error')
     else:
+        if result[0] is StepResult.LIMITED_RETRY:
+            # We allow further iterations to proceed as long as we haven't been
+            # waiting for a LIMITED_RETRY for N consecutive iterations.
+            current_iterations = (yield waiting.read()).get(group_id, 0)
+            if current_iterations > limited_retry_iterations:
+                yield msg('converge-limited-retry-too-long')
+                yield clean_waiting
+                yield delete_divergent_flag(tenant_id, group_id, version)
```

Comment:
sorry, you're right, I'll fix that. Yeah, the boundary between them stinks, as does the factoring of the whole module :(